test/python/circuit/library/test_overlap.py

```diff
@@ -131,6 +131,21 @@ def test_mismatching_qubits(self):
         with self.assertRaises(CircuitError):
             _ = UnitaryOverlap(unitary1, unitary2)
 
+    def test_insert_barrier(self):
+        """Test inserting barrier between circuits"""
+        unitary1 = EfficientSU2(1, reps=1)
+        unitary2 = EfficientSU2(1, reps=1)
+        overlap = UnitaryOverlap(unitary1, unitary2, insert_barrier=True)
+        self.assertEqual(overlap.count_ops()["barrier"], 1)
+        self.assertEqual(
+            str(overlap).strip(),
+            """
+   ┌───────────────────────────────────────┐ ░ ┌──────────────────────────────────────────┐
+q: ┤ EfficientSU2(p1[0],p1[1],p1[2],p1[3]) ├─░─┤ EfficientSU2_dg(p2[0],p2[1],p2[2],p2[3]) ├
+   └───────────────────────────────────────┘ ░ └──────────────────────────────────────────┘
+""".strip(),
+        )
+
```

Comment:
You might wanna double check this check. The `QuantumCircuit` might be displayed differently depending on where the `.draw()` command is ran. I tested this using the usual unit testing tool and you are right, it passes as normal.

However, when I run this on a python notebook, this is how the circuit is displayed:
```
┌───────────────────────────────────────┐ ░ »
q: ┤ EfficientSU2(p1[0],p1[1],p1[2],p1[3]) ├─░─»
   └───────────────────────────────────────┘ ░ »
«   ┌──────────────────────────────────────────┐
«q: ┤ EfficientSU2_dg(p2[0],p2[1],p2[2],p2[3]) ├
«   └──────────────────────────────────────────┘
```

Instead of how it is displayed on regular tests (This was run on a 16" MacBook Pro)

```
   ┌───────────────────────────────────────┐ ░ ┌──────────────────────────────────────────┐
q: ┤ EfficientSU2(p1[0],p1[1],p1[2],p1[3]) ├─░─┤ EfficientSU2_dg(p2[0],p2[1],p2[2],p2[3]) ├
   └───────────────────────────────────────┘ ░ └──────────────────────────────────────────┘
```
You should do this instead:
```suggestion
"""Test inserting barrier between circuits"""
        unitary1 = EfficientSU2(1, reps=1)
        unitary2 = EfficientSU2(1, reps=1)
        overlap = UnitaryOverlap(unitary1, unitary2, insert_barrier=True)
        self.assertEqual(overlap.count_ops()["barrier"], 1)
        self.assertEqual(
            str(overlap.draw(fold=-1, output="text")).strip(),
            """
   ┌───────────────────────────────────────┐ ░ ┌──────────────────────────────────────────┐
q: ┤ EfficientSU2(p1[0],p1[1],p1[2],p1[3]) ├─░─┤ EfficientSU2_dg(p2[0],p2[1],p2[2],p2[3]) ├
   └───────────────────────────────────────┘ ░ └──────────────────────────────────────────┘
""".strip(),
        )
```
This appears to fix the evaluation issue.

Reply:
Thank you, done!